src/gt4py/_core/filecache.py

```diff
@@ -0,0 +1,51 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
+import os
+import pathlib
+import pickle
+from typing import Any, Hashable
+
+from gt4py._core import locking
+from gt4py.eve import utils as eve_utils
+
+
+class FileCache:
+    path: pathlib.Path
```

Comment:
Missing docstring here and/or in the module level (at least one of them is needed).